File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 3/31/2026, 9:39:50 PM
class Solution {
public:
bool isPalindrome(string s){
    int n = s.size();
    int l = 0, u = n-1;
    while (l<u){
        while(l<u && !isalnum(s[l])){
            l++;
        }
        while(l<u && !isalnum(s[u])){
            u--;
        }
        if (tolower(s[l]) != tolower(s[u])){
            return false;
        }
        l++;
        u--;
    }
    return true;
}
};

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 3/31/2026, 9:39:45 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        low = 0
        high = len(s) - 1
        while low<high:
            while(not s[low].isalnum() and low<high):
                low+=1
            while(not s[high].isalnum() and low<high):
                high-=1
            if (s[low].lower()!=s[high].lower()):
                return False
            low+=1
            high-=1
        return True